tsfc/geometric.py

```diff
@@ -98,13 +98,25 @@ def translate_cell_edge_vectors(terminal, mt, params):
 
 @translate.register(CellCoordinate)
 def translate_cell_coordinate(terminal, mt, params):
-    return gem.partial_indexed(params.index_selector(lambda i: gem.Literal(params.entity_points[i]),
-                                                     mt.restriction),
-                               (params.point_index,))
+    ps = params.point_set
+    if params.integration_dim == params.fiat_cell.get_dimension():
+        return ps.expression
+
+    # This destroys the structure of the quadrature points, but since
+    # this code path is only used to implement CellCoordinate in facet
+    # integrals, hopefully it does not matter much.
+    point_shape = tuple(index.extent for index in ps.indices)
```

Comment:
Can we think of a test case where doing so would mean that evaluating `CellCoordinate` would become the leading-order cost?